pkg/apis/serving/v1alpha1/revision_validation.go

```diff
@@ -118,24 +131,74 @@ func ValidateContainerConcurrency(cc RevisionContainerConcurrencyType, cm Revisi
 	return nil
 }
 
-func validateContainer(container corev1.Container) *apis.FieldError {
+func validateVolume(volume corev1.Volume) *apis.FieldError {
+	var errs *apis.FieldError
+	if volume.Name == "" {
+		errs = apis.ErrMissingField("name")
+	}
+	vs := volume.VolumeSource
+	switch {
+	case vs.Secret != nil:
+	case vs.ConfigMap != nil:
+		// These are fine.
+	default:
+		errs = errs.Also(apis.ErrMissingOneOf("secret", "configMap"))
+	}
+	return errs
+}
+
+func validateContainer(container corev1.Container, volumes sets.String) *apis.FieldError {
 	if equality.Semantic.DeepEqual(container, corev1.Container{}) {
 		return apis.ErrMissingField(apis.CurrentField)
 	}
+
+	// Check that volume mounts match names in "volumes", that "volumes" has 100%
+	// coverage, and the field restrictions.
+	seen := sets.NewString()
+	var errs *apis.FieldError
+	for i, vm := range container.VolumeMounts {
+		// This effectively checks that Name is non-empty because Volume name must be non-empty.
+		if !volumes.Has(vm.Name) {
+			errs = errs.Also(
+				apis.ErrInvalidValue(vm.Name, "name").ViaFieldIndex("volumeMounts", i))
+		}
+		seen.Insert(vm.Name)
+
+		if vm.MountPath == "" {
+			errs = errs.Also(apis.ErrMissingField("mountPath").ViaFieldIndex("volumeMounts", i))
+		}
+		if !vm.ReadOnly {
+			errs = errs.Also(apis.ErrMissingField("readOnly").ViaFieldIndex("volumeMounts", i))
```

Comment:
Do we care, since, we already marked all the volumes as read-only. Alternatively, will k8s allow this podspec (as in they already check, do we have to?)

Reply:
We default before we validate, so this *shouldn't* hit, but I wanted to be safe vs. sorry :)